src/coreclr/jit/emitxarch.cpp

```diff
@@ -87,7 +87,23 @@ bool emitter::IsAvx512OnlyInstruction(instruction ins)
 
 bool emitter::IsApxOnlyInstruction(instruction ins)
 {
-    return (ins >= FIRST_APX_INSTRUCTION) && (ins <= LAST_APX_INSTRUCTION);
+#ifdef TARGET_AMD64
+    if (IsApxZuCompatibleInstruction(ins))
```

Comment:
nit: Also not necessarily for this PR, but ideally we'd either use a flag or a singular instruction range. Having to check many different ranges is an unnecessary negative impact to throughput.